src/lib.rs

```diff
@@ -389,14 +410,30 @@ impl Builder {
 		)
 		.expect("Failed to set up on-chain wallet");
 
-		let tx_sync = Arc::new(EsploraSyncClient::new(
-			config.esplora_server_url.clone(),
-			Arc::clone(&logger),
-		));
-
-		let blockchain =
-			EsploraBlockchain::from_client(tx_sync.client().clone(), BDK_CLIENT_STOP_GAP)
-				.with_concurrency(BDK_CLIENT_CONCURRENCY);
+		let (blockchain, tx_sync) = if let Some(chain_data_source_config) =
+			&*self.chain_data_source_config.read().unwrap()
+		{
+			match chain_data_source_config {
```

Comment:
This may be more readable if you combine the `if let` and `match` into a single match:

```rust
let (blockchain, tx_sync) = match &*self.chain_data_source_config.read().unwrap() {
	Some(ChainDataSourceConfig::Esplora(server_url)) => {
	},
	None => {
	},
};